### PR TITLE
Fix auto advance to support decimals and create a common utility for the same

### LINKED ISSUE #11643: Support any valid duration of time for auto-advance-after and animation attributes
Right now, the `auto-advance-after` attribute does not accept decimal durations or uppercase units.  We should write a common utility for parsing these durations and use it for `auto-advance-after`, `animate-in-delay`, and `animate-in-duration`.

### PR DESCRIPTION
**Pull request by prateekbh**
Fixes #11643 

- Takes out `timeStrToMillis` to a separate util
- Supports case insensitivity
- Supports decimal value for `auto-advance-after`

cc: @alanorozco @newmuis @choumx

### CODE REVIEW

File: extensions/amp-story/0.1/test/test-utils.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {timeStrToMillis} from '../utils';
+
+describes.fakeWin('amp-story utils', {}, () => {
+  describe('timeStrToMillis', () => {
+    it('should return millis for a string', () => {
```
Comment (newmuis) [inline extensions/amp-story/0.1/test/test-utils.js]: Maybe separate these into two cases?

Also I guess we should add a test for the decimal case

File: extensions/amp-story/0.1/utils.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+/**
+ * Returns millis as number if given a string(e.g. 1s, 200ms etc)
+ * @param {string} time
+ * @return {number|undefined}
+ */
+export function timeStrToMillis(time) {
+  const match = time.toLowerCase().match(/^([0-9\.]+)\s*(s|ms)$/);
+  if (!match) {
+    return;
```
Comment (choumx) [inline extensions/amp-story/0.1/utils.js]: Return NaN instead.

File: extensions/amp-story/0.1/amp-story-page.js [hunk 3]
```diff
@@ -520,12 +512,7 @@ export class AmpStoryPage extends AMP.BaseElement {
       return;
     }
 
-    let delayMs;
-    if (TIME_REGEX.MILLISECONDS.test(autoAdvanceAfter)) {
-      delayMs = Number(TIME_REGEX.MILLISECONDS.exec(autoAdvanceAfter)[1]);
-    } else if (TIME_REGEX.SECONDS.test(autoAdvanceAfter)) {
-      delayMs = Number(TIME_REGEX.SECONDS.exec(autoAdvanceAfter)[1]) * 1000;
-    }
+    const delayMs = timeStrToMillis(autoAdvanceAfter);
```
Comment (choumx) [inline extensions/amp-story/0.1/amp-story-page.js]: Nice.

File: extensions/amp-story/0.1/test/test-utils.js (hunk 1 above)
Comment (prateekbh) [reply extensions/amp-story/0.1/test/test-utils.js]: Done

File: extensions/amp-story/0.1/utils.js (hunk 2 above)
Comment (prateekbh) [reply extensions/amp-story/0.1/utils.js]: Done

**Pull request merged**